app.rb

```diff
@@ -1,7 +1,53 @@
 require 'sinatra/base'
+require_relative 'lib/player'
+require_relative 'lib/game'
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+  enable :sessions
+
+  get '/' do
+    erb :index
+  end
+
+  post '/names' do
+    player1 = Player.new(params[:player_1_name])
+    player2 = Player.new(params[:player_2_name])
+    Game.create(player1, player2, false)
+    redirect '/game'
+  end
+
+  post '/single_player' do
+    player1 = Player.new(params[:player_1_name])
+    player2 = Player.new("AI")
+    Game.create(player1, player2, true)
+    redirect '/game'
+  end
+
+  get '/game' do
+    @game = Game.instance
+    @player_1_name = @game.players[0].name
+    @player_2_name = @game.players[1].name
+    @single_player = @game.single_player
+    erb :game
+  end
+
+  post '/submit_moves' do
+    if Game.instance.single_player
+      # Tried using .sample didn't want to work idk why...
```

Comment:
We talked about this over the call, sample wasn't used in this case due to an error which stopped progress. I think it's been thoughtfully handled and corrected!